src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerUtils.cs

```
@@ -363,9 +353,9 @@ public static Size GetAdornmentDimensions(AdornmentType adornmentType)
 
     public static bool UseSnapLines(IServiceProvider provider)
     {
-        bool useSnapLines = true;
         object optionValue = null;
-        if (provider.GetService(typeof(DesignerOptionService)) is DesignerOptionService options)
+        DesignerOptionService options = provider.GetService<DesignerOptionService>();
+        if (options is not null)
```

Consider using TryGetService here

---

I deliberately didn't use it here because it would be a behaviour change. Today with throw a NRE if `provider` is null. If I make the change to `TryGetService`, the method returns `true` for a null `provider`

---

Could you please throw a better exception if provider is null, and then use the TryGet method?